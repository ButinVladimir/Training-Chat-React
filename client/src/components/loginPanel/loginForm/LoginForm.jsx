import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

function LoginForm({ login, onChangeLogin, onLogin }) {
  return (
    <div className="form-row">
      <input type="text" className="input" value={login} onChange={e => onChangeLogin(e.target.value)} />
      <button type="button" className="login" onClick={onLogin}>Login</button>
    </div>
  );
}

LoginForm.propTypes = {
  login: PropTypes.string.isRequired,
  onChangeLogin: PropTypes.func.isRequired,
  onLogin: PropTypes.func.isRequired,
};

export default LoginForm;
