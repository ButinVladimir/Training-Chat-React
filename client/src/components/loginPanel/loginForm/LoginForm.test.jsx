import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', () => {
  it('renders without crashing', () => {
    const component = shallow(
      <LoginForm
        login="login"
        onChangeLogin={() => {}}
        onLogin={() => {}}
      />,
    );
    expect(component).toMatchSnapshot();
  });

  it('handles changing login', () => {
    const mock = jest.fn();

    const component = shallow(
      <LoginForm
        login="login"
        onChangeLogin={mock}
        onLogin={() => {}}
      />,
    );

    const input = component.find('input');
    input.value = 'newLogin';
    input.simulate('change', { target: { value: 'value' } });

    expect(mock).toHaveBeenCalledTimes(1);
    expect(mock).toHaveBeenCalledWith('value');
  });

  it('handles click on login button correctly', () => {
    const mock = jest.fn();

    const component = shallow(
      <LoginForm
        login="login"
        onChangeLogin={() => {}}
        onLogin={mock}
      />,
    );

    component.find('button').simulate('click');
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
