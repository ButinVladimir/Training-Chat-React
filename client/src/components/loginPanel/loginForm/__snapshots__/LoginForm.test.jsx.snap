// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm renders without crashing 1`] = `
<div
  className="form-row"
>
  <input
    className="input"
    onChange={[Function]}
    type="text"
    value="login"
  />
  <button
    className="login"
    onClick={[Function]}
    type="button"
  >
    Login
  </button>
</div>
`;
